mypy/binder.py

```diff
@@ -189,19 +197,20 @@ def pop_frame(self, can_skip: bool, fall_through: int) -> Frame:
 
         return result
 
-    def get_declaration(self, expr: Node) -> Type:
+    def get_declaration(self, expr: BindableExpression) -> Type:
+        assert not isinstance(expr, SymbolTableNode)
```

Comment:
This assert can't trigger given the definition of BindableExpression right? Or am I misunderstanding?